### PR TITLE
Various fix for "ng test"

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
Corrects many nonsense things.

### CODE REVIEW

File: packages/@angular/cli/commands/test.ts [hunk 1]
```diff
@@ -26,9 +26,8 @@ export interface TestOptions {
 const TestCommand = EmberTestCommand.extend({
   availableOptions: [
     {
-      name: 'watch',
+      name: 'auto-watch',
       type: Boolean,
-      default: true,
       aliases: ['w'],
```
Comment (delasteve) [inline packages/@angular/cli/commands/test.ts]: Add `auto-watch` to this array.

**Comment by ishitatsuyuki** [comment]
Due to how it pass options to Karma the "--watch" option was meaningless before. It looks like breaking change, but actually​ it never worked before. I can add back the alias if that's preferred.

**Comment by delasteve** [comment]
I'm a little confused on what part is meaningless. I am able to do the following and they all work as expected

Watch mode mode
```
ng test
ng test --watch
ng test --single-run=false
``

Single run mode
``
ng test --no-watch
ng test --watch=false
ng test --single-run
```

Removing `--watch` is a breaking change. It is a flag that's in there. Removing it breaks current functionality (think about a CI script that uses `ng test --watch=false` or `ng test --no-watch`).

**Comment by ishitatsuyuki** [comment]
I'm planning to bring the --watch discussion elsewhere. It was actually a coincidence, and "--watch=false" and "--single-run" basically did the same thing.

Will update this with the change of defaults only.

**Comment by ishitatsuyuki** [comment]
@delasteve #5897

**Comment by ishitatsuyuki** [comment]
The breaking changes are removed, and now the purpose is to honor the values set in karma.conf.js. With the defaults, they're forcefully overridden even when no flags are specified. I think the commit message explains well.

**Pull request merged**